clients/src/main/java/org/apache/kafka/common/errors/ResourceNotReadyException.java

```diff
@@ -0,0 +1,14 @@
+package org.apache.kafka.common.errors;
+
+public class ResourceNotReadyException extends RetriableException {
```

Comment:
Sorry, `org.apache.kafka.common.errors` is a public API, we need a KIP to add it. That's why I suggested to use the existing `ReplicaNotAvailableException` instead. Could we change it and we can write a KIP for this new `ResourceNotReadyException` if you think it's more appropriate?

Reply:
+1 on not introducing a new error as it requires a KIP. 
We can reuse the closest possible error for now.  `ReplicaNotAvailableException` triggers refreshing a metadata update from the client as it is kind of an `InvalidMetadataException` which may generate a lot of metadata calls. It may be good to return an exception that does not call for metadata refresh.

We can introduce a new error as a separate KIP later and handle it appropriately.

Reply:
We couldn't find another exception that fits our case. We can use `ReplicaNotAvailableException` for now. I have created a jira to introduce a new retryable exception/error code.
https://issues.apache.org/jira/browse/KAFKA-15405

Reply:
That looks reasonable to me for now.

Reply:
Updated the PR.